### PR TITLE
chore: add a zzz-release binary that update the version of crates

### PR DESCRIPTION
**Pull request by hansl**
It updates the versions, create a commit, and show which packages need to be published in what order.

### CODE REVIEW

**Comment by hansl** [comment]
This does not follow best practices, sorry. I can split some functions so it becomes easier to read.

File: zzz-release/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+    /// Skip creating a git commit (will still output all messages and logs).
+    #[clap(long = "skip-git")]
+    skip_git: bool,
+
+    /// If true, will ignore status and branch of the repo.
+    #[clap(long = "force")]
+    force: bool,
+
+    /// The last release tag, by default the last tag is used if it starts
+    /// with `release-`.
```
Comment (ericswanson-dfinity) [inline zzz-release/src/main.rs:25]:
```suggestion
    /// The release tag, by default the most recent tag is used if it starts
    /// with `release-`.
```

**Comment by ericswanson-dfinity** [comment]
Let's have in the readme a section that tells what we should `cargo run` when releasing

File: zzz-release/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+fn update_manifest(package_name: &str, version_map: &VersionMap) -> Result<Box<ApplyChangeFn>> {
+    let (package, version) = &version_map.get(package_name).unwrap();
+    eprintln!("   {:20} {} => {}", package_name, package.version, version);
+    let cargo_path = package.manifest_path.as_path().as_os_str().to_os_string();
+    let mut cargo_toml = std::fs::read_to_string(&cargo_path)?;
+
+    let parsed = toml_parse::parse_it(&cargo_toml)?;
+    let root = parsed.syntax();
+
+    let mut changes: Vec<(usize, usize, String)> = Vec::new();
```
[the remark this answers is not in the document]
Comment (hansl) [reply zzz-release/src/main.rs]: `RangeBounds` isn't the right one (it's a trait not a struct), but `Range` works!

File: zzz-release/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+                    .unwrap()
+                    .first_token()
+                    .unwrap()
+                    .next_sibling_or_token()
+                    .unwrap();
+                Some((key, value))
+            }
+        })
+        .take(1)
+        .for_each(|(_k, v)| {
```
[the remark this answers is not in the document]
Comment (hansl) [reply zzz-release/src/main.rs]: Changed that.

File: zzz-release/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+    let parsed = toml_parse::parse_it(&cargo_toml)?;
+    let root = parsed.syntax();
+
+    let mut changes: Vec<(usize, usize, String)> = Vec::new();
+
+    // Find the `[package]` object and the key pair for version = ...,
+    // then add a change to the vector above to update the value to the new version.
+    walk(&root)
+        .filter_map(|element| {
+            if element.kind() != TomlKind::KeyValue || element.parent().is_none() {
```
[the remark this answers is not in the document]
Comment (hansl) [reply zzz-release/src/main.rs]: Let me know what you think of the latest changes.

**Review by ericswanson-dfinity — APPROVED** [review]
There's room to refactor the larger methods into smaller ones, but I can take that on while you're out.

**Pull request merged**